share/wake/lib/core/json.wake

```diff
@@ -538,3 +538,108 @@ export def lastValueInList (values: List JValue): Result JValue Error =
             | flatten
             | deduplicateJObjectKeys lastValueInList
             | rmap JObject
+
+export def jString: JValue => List String = match _
+    JString s = |> s
+    _         = Nil
+
+export def editJString (f: String => String): JValue => JValue = match _
+    JString s = JString (f s)
+    x         = x
+
+export def jInteger: JValue => List Integer = match _
+    JInteger i = |> i
+    _          = Nil
+
+export def editJInteger (f: Integer => Integer): JValue => JValue = match _
+    JInteger i = JInteger (f i)
+    x          = x
+
+export def jBoolean: JValue => List Boolean = match _
+    JBoolean b = |> b
+    _          = Nil
+
+export def editJBoolean (f: Boolean => Boolean): JValue => JValue = match _
+    JBoolean b = JBoolean (f b)
+    x          = x
+
+export def jDouble: JValue => List Double = match _
+    JDouble d = |> d
+    _         = Nil
+
+export def editJDouble (f: Double => Double): JValue => JValue = match _
+    JDouble d = JDouble (f d)
+    x         = x
+
+export def jNumber: JValue => List Double = match _
+    JDouble x  = |> x
+    JInteger x = |> x | mapPartial dint
+    _          = Nil
+
+export def editJNumber (f: Double => Double): JValue => JValue = match _
+    JDouble x  = JDouble (f x)
+    JInteger x = match (dint x)
+        Some dx = JDouble (f dx)
+        None    = JInteger x
+    x          = x
+
+export def jNull: JValue => List Unit = match _
+    JNull = |> Unit
+    _     = Nil
+
+export def editJNull (f: Unit => Unit): JValue => JValue = match _
```

Comment:
Hmm. I'm not sure what to make of this one. ;-)

Reply:
Yeah its a bit funky but its there for symmetry. Like imagine if we didn't have the "id" function, the id function seems really silly but you need it sometimes for composing with other things.

Also keep in mind that functions have effects and you might want to perform an *effect*  on a null JValue.

Reply:
I agree we should have it.

FYI, we do NOT have the `id` function in the wake standard library. ;)
... because we have `(_)`.